### PR TITLE
Add first test, simply testing a buffer

### PR DESCRIPTION
**Pull request by mvdan**
Updates #3.

### CODE REVIEW

File: Tests.hs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    | otherwise = do
+        printf "OK "
+        print want
+
+data Signal = A | B deriving (Eq, Show, Enum, Bounded)
+
+testBuffer = do
+    putStrLn "=== testBuffer"
+    assertEq True $ initial circuit initState
+    assertEq [Transition A True] $ enabledTransitions initState circuit
```
Comment (snowleopard) [inline Tests.hs]: I think the following would be a bit more readable?

``` haskell
assertEq [rise A] $ enabledTransitions initState circuit
let afterA = fire (rise A) initState
assertEq [fall A, rise B] $ enabledTransitions afterA circuit
```

**Comment by snowleopard** [comment]
@mvdan Looks good, just one comment above.

File: Tests.hs (hunk 1 above)
Comment (mvdan) [inline Tests.hs]: Done, thanks for the suggestion! I knew it looked way too verbose.

**Comment by snowleopard** [comment]
:+1:

**Pull request merged**